mne/io/artemis123/artemis123.py

```diff
@@ -0,0 +1,258 @@
+
+    ref_grad_names = ['REF_007', 'REF_008', 'REF_009',
+                      'REF_010', 'REF_011', 'REF_012']
+
+    # load mne loc dictionary
+    loc_dict = _load_mne_locs()
+    info['chs'] = []
+    info['bads'] = []
+
+    for i, chan in enumerate(header_info['channels']):
+        # build chs struct
+        t = {'cal': float(chan['scaling']), 'ch_name': chan['name'],
+             'logno': i + 1, 'scanno': i + 1, 'range': 1.0,
+             'unit_mul': FIFF.FIFF_UNITM_NONE,
+             'coord_frame': FIFF.FIFFV_COORD_DEVICE}
+        try:
+            t['loc'] = loc_dict[chan['name']]
+        except:
+            t['loc'] = np.zeros(12)
+
+        if (chan['name'].startswith('MEG')):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_GRAD
+            t['kind'] = FIFF.FIFFV_MEG_CH
+            # TODO Shouldn't gradiometers be in T/m (Not sure if it matters)
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # 3 axis referance magnetometers
+        elif (chan['name'] in ref_mag_names):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_REF_MAG
+            t['kind'] = FIFF.FIFFV_REF_MEG_CH
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # reference gradiometers
+        elif (chan['name'] in ref_grad_names):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_REF_GRAD
+            t['kind'] = FIFF.FIFFV_REF_MEG_CH
+            # TODO Shouldn't gradiometers be in T/m (Not sure if it matters)
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # other reference channels are unplugged and should be ignored.
+        elif (chan['name'].startswith('REF')):
+            t['coil_type'] = FIFF.FIFFV_COIL_NONE
+            t['kind'] = FIFF.FIFFV_MISC_CH
+            t['unit'] = FIFF.FIFF_UNIT_V
+            info['bads'].append(t['ch_name'])
+
+        elif (chan['name'].startswith('AUX')):
+            t['coil_type'] = FIFF.FIFFV_COIL_NONE
+            t['kind'] = FIFF.FIFFV_MISC_CH
+            t['unit'] = FIFF.FIFF_UNIT_V
+
+        elif (chan['name'].startswith('TRG')):
+            t['coil_type'] = FIFF.FIFFV_COIL_NONE
+            t['kind'] = FIFF.FIFFV_STIM_CH
+            t['unit'] = FIFF.FIFF_UNIT_V
+
+        elif (chan['name'].startswith('MIO')):
+            t['coil_type'] = FIFF.FIFFV_COIL_NONE
+            t['kind'] = FIFF.FIFFV_MISC_CH
+            t['unit'] = FIFF.FIFF_UNIT_V
+
+        else:
+            raise ValueError('Channel does not match expected' +
+                             ' channel Types:"%s"' % chan['name'])
+
+        info['chs'].append(t)
+        if (chan['FLL_ResetLock'] == 'TRUE'):
+            info['bads'].append(t['ch_name'])
+
+    # reduce info['bads'] to unique set
+    info['bads'] = list(set(info['bads']))
+    info['ch_names'] = [ch['ch_name'] for ch in info['chs']]
+    info['nchan'] = len(info['ch_names'])
```

Comment:
Get rid of these last two, use `info._update_redundant()`. If this was copied from another reader, please update there, too...